browser/ui/views/location_bar/brave_news_location_view.cc

```diff
@@ -10,210 +10,154 @@
-    SetImage(ButtonState::STATE_NORMAL, image);
-    SetVisible(has_feeds);
   }
 
-  // views::LabelButton:
-  std::unique_ptr<views::LabelButtonBorder> CreateDefaultBorder()
-      const override {
-    std::unique_ptr<views::LabelButtonBorder> border =
-        LabelButton::CreateDefaultBorder();
-    border->set_insets(gfx::Insets::VH(3, 0));
-    return border;
+  // Don't show the icon if preferences don't allow
+  if (!tab_helper || !should_show_.GetValue() || !news_enabled_.GetValue() ||
+      !opted_in_.GetValue()) {
+    SetVisible(false);
+    return;
   }
 
-  std::u16string GetTooltipText(const gfx::Point& p) const override {
-    return l10n_util::GetStringUTF16(IDS_BRAVE_NEWS_ACTION_VIEW_TOOLTIP);
-  }
+  // Icon color changes if any feeds are being followed
+  bool subscribed = false;
+  subscribed = tab_helper->IsSubscribed();
+  SetIconColor(GetIconColor(subscribed));
 
-  SkPath GetHighlightPath() const {
-    gfx::Rect rect(GetPreferredSize());
-    const int radii = ChromeLayoutProvider::Get()->GetCornerRadiusMetric(
-        views::Emphasis::kMaximum, rect.size());
-    SkPath path;
-    path.addRoundRect(gfx::RectToSkRect(rect), radii, radii);
-    return path;
-  }
+  // Don't show icon if there are no feeds
+  const bool has_feeds = !tab_helper->GetAvailableFeeds().empty();
+  const bool is_visible = has_feeds || IsBubbleShowing();
+  SetVisible(is_visible);
+}
 
-  // TabStripModelObserver:
-  void OnTabStripModelChanged(
-      TabStripModel* tab_strip_model,
-      const TabStripModelChange& change,
-      const TabStripSelectionChange& selection) override {
-    if (selection.active_tab_changed()) {
-      if (selection.old_contents) {
-        BraveNewsTabHelper::FromWebContents(selection.old_contents)
-            ->RemoveObserver(this);
-      }
-
-      if (selection.new_contents) {
-        BraveNewsTabHelper::FromWebContents(selection.new_contents)
-            ->AddObserver(this);
-      }
-    }
+const gfx::VectorIcon& BraveNewsLocationView::GetVectorIcon() const {
+  return kBraveNewsSubscribeIcon;
+}
 
-    Update();
-  }
+std::u16string BraveNewsLocationView::GetTextForTooltipAndAccessibleName()
+    const {
+  return l10n_util::GetStringUTF16(IDS_BRAVE_NEWS_ACTION_VIEW_TOOLTIP);
+}
 
-  // BraveNewsTabHelper::PageFeedsObserver:
-  void OnAvailableFeedsChanged(
-      const std::vector<BraveNewsTabHelper::FeedDetails>& feeds) override {
-    Update();
-  }
+bool BraveNewsLocationView::ShouldShowLabel() const {
+  return false;
+}
 
-  // views::View:
-  void OnThemeChanged() override {
-    views::LabelButton::OnThemeChanged();
-    Update();
-  }
+void BraveNewsLocationView::OnAvailableFeedsChanged(
+    const std::vector<BraveNewsTabHelper::FeedDetails>& feeds) {
+  Update();
+}
 
- private:
-  void ButtonPressed() {
-    // If the bubble is already open, do nothing.
-    if (bubble_widget_) {
-      return;
-    }
+void BraveNewsLocationView::OnThemeChanged() {
+  views::LabelButton::OnThemeChanged();
```

Comment:
Should this be PageActionIconView()? Or is this intended to bypass the implementation of PageActionIconView()? If it's the latter, having a comment abou t it would be great!

Reply:
Thanks for spotting my mistake leaving this from the previous version - fixed